exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -56,50 +82,88 @@ def __init__(self,
             self.excluded_extensions = []
 
         self.path_hasher = PathHasher(hashfunc,
-                                      hash_permissions=hash_permissions,
-                                      use_relative_paths=use_relative_paths)
+                                      hash_permissions=hash_permissions)
         self.file_content_hasher = FileContentHasher(hashfunc, blocksize)
 
-    def hash(self, files_and_directories: List[str]) -> bytes:
-        collected_directories = []
-        collected_files = []
+    def hash(self, files_and_directories: List[PathMapping]) -> bytes:
         if not isinstance(files_and_directories, List):
             raise Exception("List with paths expected and not '%s' with type %s"
                             % (files_and_directories, type(files_and_directories)))
+
+        directory_mapping_result = self.collect_dest_path_and_src_files(files_and_directories)
+        hashes = self.compute_hashes(directory_mapping_result)
+        return self._reduce_hash(hashes)
+
+    def validate(self, mappings: List[DestinationMapping]) -> None:
+        # Verify that there are no duplicate mappings to same destination
+        destinations = [p.destination for p in mappings]
+        if len(set(destinations)) != len(destinations):
+            raise AssertionError(f"Directory content for hashing contains duplicates: {destinations}")
+
+    def collect_dest_path_and_src_files(self, files_and_directories: List[PathMapping]) -> DirectoryMappingResult:
+        """
+        Iterate over all mappings and collect:
+        1) Destination paths-names (for Hashing): Mapping of source -> destination of file/directory, ordered by destination path
```

Comment:
This is not completely clear, are these two different things the path names and the mapping

Reply:
This is another mapping, which maps the found source file/directory to it's destination counterpart.
You can think about the input mappings (files_and_directories) as the templates, and the one returned here as the instances.

Reply:
I changed the documentation. Hope its now more clear.